src/index.ts

```diff
@@ -293,29 +300,29 @@ async function statusOk(res: Response, logger: Logger): Promise<boolean> {
     return String(res.status)[0] === '2'
 }
 
-export function getProperties(event: PluginEvent, propertiesToInclude: string): Properties {
-    // Spreading so the TypeScript compiler understands that in the
-    // reducer there's no way the properties will be undefined
+// we allow `any` since we don't know what type the properties are, and `unknown` is too restrictive here
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getNestedProperty(properties: Record<string, any>, path: string): any {
+    return path.split('.').reduce((acc, part) => acc[part], properties);
+}
+
+export function getProperties(event: PluginEvent, propertiesToInclude: string, fieldMappings: FieldMappings = {}): Properties {
     const { properties } = event
 
     if (!properties) {
         return {}
     }
 
-    if (!propertiesToInclude?.trim()) {
-        return properties
-    }
-
-    const allParameters = propertiesToInclude.split(',')
-    const propertyKeys = Object.keys(properties)
+    // if no propertiesToInclude is set then all properties are allowed
+    const propertiesAllowList = !!propertiesToInclude?.trim().length ? propertiesToInclude.split(',').map((e) => e.trim()) : Object.keys(properties)
 
-    return allParameters.reduce<Record<string, unknown>>((acc, currentValue) => {
-        const trimmedKey = currentValue.trim()
+    const mappedProperties: Record<string, any> = {}
 
-        if (propertyKeys.includes(trimmedKey)) {
-            acc[trimmedKey] = properties[trimmedKey]
-        }
+    propertiesAllowList.forEach((allowedProperty) => {
+        const val = getNestedProperty(properties, allowedProperty)
+        const mappedKey = fieldMappings[allowedProperty] || allowedProperty
+        mappedProperties[mappedKey] = val
+    })
```

Comment:
if I'm reading this right, then we loop over allowedProperty, which is stuff like `[user, shopping_cart]` (because above we do `Object.keys(properties)`), which itself can have nested properties, e.g. `shopping_cart.item.description`. So if in fieldMappings I did `shopping_cart.item.description: 'item_description'` that wouldn't get overridden. How are nested properties supposed to work?

Reply:
not sure if I understand the question but I tried to enable mapping like this one: "fieldMappings": { "Email": "person_properties.email" }

Reply:
this is all i've tested against

https://github.com/PostHog/salesforce-plugin/pull/11/files#diff-ae645b48b3ca416f3c051080bb7c156b53f8607867a1295a8802882fb1aec015R33-R66

can you express your question as a test there maybe? 

i think

if you said `shopping_cart` is allowed we'd copy in `shopping_cart` unaltered
if you said `shopping_cart.item.description` is allowed we'd copy in `shopping_cart.item.description` _not all of shopping_cart`

for each of those if you want a field mapping you have to add it

so if you said `shopping_cart` is allowed and gave a field mapping of `shopping_cart.item.description` then you've given effectively invalid config 

we don't support (with this code) an allowed property which is an object **and** renaming field inside that object

i think what @minekansu has discovered is that we can't be sending arbitrary blobs to salesforce, so folk should always be setting an allowed properties and field mapping that are very specific

but we're always sort of guessing at what salesforce wants 🙈

Reply:
> so if you said shopping_cart is allowed and gave a field mapping of shopping_cart.item.description then you've given effectively invalid config

ok cool if that's expected then this is all good - it might be useful for the user if we debug logged this case, but also I wouldn't spend too much time improving plugins at this point

Reply:
yep, salesforce API gives very little back so we also a little bit need to get this in and validate it 🙈 :)